src/App.tsx

```diff
@@ -1,41 +1,102 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
 import './App.scss';
 import './styles/general.scss';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { Loader } from './components/Loader';
+
+import { getUserPosts } from './api/posts';
+
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [selectedUserId, setSelectedUserId] = useState<number>(0);
+  const [selectedPostId, setSelectedPostId] = useState<number>(0);
+  const [isError, setIsError] = useState<boolean>(false);
```

Comment:
you don't need generics for primitives types, they are set automatically (if state won't have any other type)
```suggestion
  const [selectedUserId, setSelectedUserId] = useState(0);
  const [selectedPostId, setSelectedPostId] = useState(0);
  const [isError, setIsError] = useState(false);
```